tls/extensions/s2n_client_key_share.c

```diff
@@ -156,6 +222,11 @@ static int s2n_client_key_share_recv(struct s2n_connection *conn, struct s2n_stu
 
         GUARD(s2n_ecc_evp_read_params_point(extension, share_size, &point_blob));
 
+        /* Ignore curves with no keyshare */
+        if (*point_blob.data == 0) {
```

Comment:
Should this be == NULL instead of 0, just for clarity?

Reply:
Actually, we require it to be 0 and not NULL, this is because we are representing empty key shares with a 0 value.  If the value was NULL it we cannot parse the point in `s2n_ecc_evp_read_params_point` prior.

Reply:
I'd put that in a comment. You might also to do a memcmp here-- I'm guessing the entire data blob needs to be zero?

Would this also trigger on a point_blob where the first byte just happens to be zero? You're dereferencing `data`, which is a uint8_t*, returning the value of first element it points to.